### PR TITLE
[IOPAE-1529] Setup ApplicationGateway for iPatente applications

### PR DESCRIPTION
**Pull request by giamma1295**
### Motivation and Context
Setup Application Gateway for  Vehicles and Licences iPatente app.

### Major Changes

- Add Vehicles and Licences A Record on iPatente dnsZone
- Create listener, route and all necessary to expose the mentioned above iPatente application through applicationGateway

### CODE REVIEW

File: src/common/_modules/application_gateway/main.tf [hunk 1]
```diff
@@ -412,18 +412,35 @@ module "app_gw" {
...
+      ssl_profile_name   = format("%s-ssl-profile", var.project)
+      firewall_policy_id = null
+
+      certificate = {
+        name = var.certificates.licences_ipatente_io_pagopa_it
+        id = replace(
+          data.azurerm_key_vault_certificate.app_gw_licences_ipatente_io.secret_id,
+          "/${data.azurerm_key_vault_certificate.app_gw_licences_ipatente_io.version}",
           ""
         )
```
Comment (christian-calabrese) [inline src/common/_modules/application_gateway/main.tf:445]: Should make the code easier with the same result
```suggestion
        id = data.azurerm_key_vault_certificate.app_gw_licences_ipatente_io.versionless_secret_id
```

File: src/common/_modules/application_gateway/main.tf [hunk 2]
```diff
@@ -412,18 +412,35 @@ module "app_gw" {
...
       ssl_profile_name   = format("%s-ssl-profile", var.project)
       firewall_policy_id = null
 
       certificate = {
-        name = var.certificates.ipatente_io_pagopa_it
+        name = var.certificates.vehicles_ipatente_io_pagopa_it
         id = replace(
-          data.azurerm_key_vault_certificate.app_gw_ipatente_io.secret_id,
-          "/${data.azurerm_key_vault_certificate.app_gw_ipatente_io.version}",
+          data.azurerm_key_vault_certificate.app_gw_vehicles_ipatente_io.secret_id,
```
Comment (christian-calabrese) [inline src/common/_modules/application_gateway/main.tf:425]: Ditto (versionless_secret_id)

File: src/common/_modules/application_gateway/main.tf [hunk 3]
```diff
@@ -412,18 +412,35 @@ module "app_gw" {
...
+          "/${data.azurerm_key_vault_certificate.app_gw_vehicles_ipatente_io.version}",
```
[the remark this answers is not in the document]
Comment (giamma1295) [reply src/common/_modules/application_gateway/main.tf]: done in 4dea3c4778343002e72a4c254af1358b9f3e1026

File: src/common/_modules/application_gateway/main.tf [hunk 4]
```diff
@@ -868,49 +867,43 @@ module "app_gw" {
...
-          }]
-          url = {
-            path         = "/{var_uri_path_1}"
-            query_string = null
-            reroute      = false
-            components   = "path_only"
-          }
-          request_header_configurations  = []
-          response_header_configurations = []
+      name = "rewrite-rule-set-vehicles-ipatente-io-app"
```
Comment (Krusty93) [inline src/common/_modules/application_gateway/main.tf:870]: Can we use `local.io_backend_ip_headers_rule`?

File: src/common/_modules/application_gateway/main.tf [hunk 5]
```diff
@@ -868,49 +867,43 @@ module "app_gw" {
...
-          }]
-          url = {
-            path         = "/{var_uri_path_1}"
-            query_string = null
-            reroute      = false
-            components   = "path_only"
-          }
-          request_header_configurations  = []
-          response_header_configurations = []
+      name = "rewrite-rule-set-licences-ipatente-io-app"
```
Comment (Krusty93) [inline src/common/_modules/application_gateway/main.tf:890]: ditto

File: src/common/_modules/application_gateway/main.tf [hunk 6]
```diff
@@ -503,6 +512,20 @@ module "app_gw" {
...
+      listener              = "vehicles-ipatente-io-pagopa-it"
+      backend               = "vehicles-ipatente-io-app"
+      rewrite_rule_set_name = "rewrite-rule-set-vehicles-ipatente-io-app"
+      priority              = 130
+    }
+
+    licences-ipatente-io-pagopa-it = {
+      listener              = "licences-ipatente-io-pagopa-it"
+      backend               = "licences-ipatente-io-app"
+      rewrite_rule_set_name = "rewrite-rule-set-licences-ipatente-io-app"
```
Comment (Krusty93) [inline src/common/_modules/application_gateway/main.tf:526]: Can we use `local.io_backend_ip_headers_rule`?

File: src/common/_modules/application_gateway/main.tf [hunk 7]
```diff
@@ -503,6 +512,20 @@ module "app_gw" {
       rewrite_rule_set_name = "rewrite-rule-set-fims-op-app"
       priority              = 120
     }
+
+    vehicles-ipatente-io-pagopa-it = {
+      listener              = "vehicles-ipatente-io-pagopa-it"
+      backend               = "vehicles-ipatente-io-app"
+      rewrite_rule_set_name = "rewrite-rule-set-vehicles-ipatente-io-app"
```
Comment (Krusty93) [inline src/common/_modules/application_gateway/main.tf:519]: ditto

File: src/common/_modules/application_gateway/main.tf [hunk 8]
```diff
@@ -412,20 +412,29 @@ module "app_gw" {
       }
     }
 
-    ipatente-io-pagopa-it = {
+    vehicles-ipatente-io-pagopa-it = {
```
Comment (Krusty93) [inline src/common/_modules/application_gateway/main.tf:415]: Can we add this change in the next PR? Just to have a Terraform plan easier to read

File: src/common/_modules/application_gateway/main.tf (hunk 5 above)
Comment (giamma1295) [reply src/common/_modules/application_gateway/main.tf]: 8f8ecd8c02bb0fdf9613ae58dd7bffe509f8e41b

File: src/common/_modules/application_gateway/main.tf (hunk 6 above)
Comment (giamma1295) [reply src/common/_modules/application_gateway/main.tf]: 8f8ecd8c02bb0fdf9613ae58dd7bffe509f8e41b

File: src/common/_modules/application_gateway/main.tf (hunk 7 above)
Comment (giamma1295) [reply src/common/_modules/application_gateway/main.tf]: 8f8ecd8c02bb0fdf9613ae58dd7bffe509f8e41b

**Review by Krusty93 — APPROVED** [review]

**Pull request merged**